### DISCUSSION TITLE
How to load a checkpoint and use the visualization tools without redefining the problem object

### DISCUSSION THREAD

**Discussion by Leonidas-Z** [anyoptimization/pymoo#583] [Q&A]
I am running Pymoo on a cluster and I am calculating the objective function using cluster modules that I cannot install on my personal computer. 

What I want to achieve is to transfer from the cluster to my personal computer a checkpoint binary file, and use this to do post-processing (visualization, data manipulation, etc), on my personal computer. 

Since the problem object needs modules that are installed on the cluster, I am not able to initialize the result=minimize(problem, checkpoint) to be able to utilize the visualization tools and have access to history data.

What should I do?

**Comment by blankjul** [comment]
You could probably use a pattern to solve this issue. The flyweight pattern might be one that works well. The problem is the flyweight and then you can from there return the full object that implements the evaluation.
